service/history/api/workflow_id_dedup.go

```diff
@@ -109,11 +111,17 @@ func ResolveDuplicateWorkflowID(
 func resolveDuplicateWorkflowStart(
 	shardContext shard.Context,
 	currentWorkflowStartTime time.Time,
+	workflowKey definition.WorkflowKey,
 	newRunID string,
-	workflowID string,
 ) (UpdateWorkflowActionFunc, error) {
 
-	minimalReuseInterval := shardContext.GetConfig().WorkflowIdReuseMinimalInterval()
+	namespaceEntry, err := GetActiveNamespace(shardContext, namespace.ID(workflowKey.NamespaceID))
+	if err != nil {
+		return nil, err
+	}
+	nsName := namespaceEntry.Name().String()
+	minimalReuseInterval := shardContext.GetConfig().WorkflowIdReuseMinimalInterval(nsName)
```

Comment:
I think both callers of ResolveDuplicateWorkflowID() already have access to namespaceEntry. We don't need the extra ns registry look up and conversion from ID to name. 
Method can take in namespace.Namespace and commonpb.WorkflowExecution

Reply:
I checked, it is called from 
func createWorkflowMutationFunction(
which doesn't have (at least direct) access to namespace entry.

If getting namespace is heavy operation (I do see some lock/unlock), we should consider adding namespace entry to something like workflow context (instead of just namespace id).
Or use config by namespace id.

Because as far as. I can tell "have namespace id-> get namespace name->get some config value" is a common way of using namespace name

Reply:
For createWorkflowMutationFunction you can it via currentMutableState.GetNamespaceEntry()

Reply:
I don't have very strong opinion on the change though. The look up is not that heavy compared to the persistence operation.

Reply:
I find a way to provide it